cmake/functions.cmake

```diff
@@ -148,3 +151,14 @@ function(JOIN VALUES GLUE OUTPUT)
   string (REGEX REPLACE "[\\](.)" "\\1" _TMP_STR "${_TMP_STR}") #fixes escaping
   set (${OUTPUT} "${_TMP_STR}" PARENT_SCOPE)
 endfunction()
+
+# if compiler supports flag FLAG1, then OUT=FLAG1, else OUT=FLAG2
+function(test_flag_support_or_fallback OUT FLAG1 FLAG2)
+  include(CheckCXXCompilerFlag)
+  check_cxx_compiler_flag(${FLAG1} has_${FLAG1})
```

Comment:
Same check should be applied to second flag.

Reply:
Why? It is a shortcut for 
`OUT=defined(flag1) ? flag1 : flag2`

By definition, flag2 should be supported